### ISSUE TITLE
Visual Studio 2017 support

### ISSUE THREAD

**Issue by HeikoStudt** [pasztorpisti/vs-window-title-changer#10]
Hi @pasztorpisti,

it would be nice to update the extension for adding Visual Studio 2017 support as you did for 2015:
[Commit for VS 2015](https://github.com/pasztorpisti/vs-window-title-changer/commit/bf21ed21cca915d444a0540ae668ed10e51ae07c)

Many Thanks,
MFG
HeikoStudt

**Comment by pasztorpisti** [comment]
Hi @HeikoStudt,

Unfortunately I no longer do windows development and lack the time to buy/install windows along with whole dev environment to implement and test this change.

If you have time to test the simplest route (manifest update for VS2017) and it works then a pull request (similar to the previous change/update you've linked) would be welcome.

The simplest way to test the update for VS2017 (this is how I've done the previous update you linked):
- Download the latest binary version of the plugin which is stored here in an odd way for historic reasons: https://github.com/pasztorpisti/vs-window-title-changer/blob/master/Builds/r112/VSWindowTitleChanger.vsix
- The .vsix plugin is in zip format: extract it
- Change the extracted `extension.vsixmanifest` file by adding an entry for VS2017
- Re-zip the extracted/modified plugin into the new shiny VS2017 compatible .vsix file
- Try to install the vsix with VS2017 by clicking the updated .vsix file

What you have to add to the extracted `extension.vsixmanifest` is very likely to be:
```xml
      <VisualStudio Version="15.0">
        <Edition>Community</Edition>
        <Edition>Enterprise</Edition>
        <Edition>Pro</Edition>
        <Edition>IntegratedShell</Edition>
        <Edition>VSLS</Edition>
      </VisualStudio>
```

If it works then the manifest xml changes should be added to the `source.extension.vsixmanifest` file of this repo. A pull request with the updated vsix file and the updated manifest (`Builds/r116/VSWindowTitleChanger.vsix`, `source.extension.vsixmanifest`) would be great.

Even if the change works, it is expected to show warnings during installation because the correct update for VS2017 would require a rebuild with the new VS SDK: http://www.visualstudioextensibility.com/2017/01/10/its-time-to-change-the-vsix-manifest-of-your-extension-to-v3-for-visual-studio-2017-compatibility/

Cheers,
pasztorpisti

**Comment by mdvtj** [comment]
I am currently doing researches for adding VS2017 support to this **great extension**. This time (VS2015 -> VS2017) a little bit more work is required, [as Carlos Quintero points out](http://www.visualstudioextensibility.com/2017/01/10/its-time-to-change-the-vsix-manifest-of-your-extension-to-v3-for-visual-studio-2017-compatibility/).

As it turns out, VS2017 does not support any more version 1 of the vsixmanifest file. As Carlos explains, this effectively means to distribute two separate extensions or drop support for VS2010 (which would make this process way more straightforward).

@pasztorpisti, do you mind dropping support for VS2010 in future versions of this extension?

**Comment by HeikoStudt** [comment]
The change @pasztorpisti suggested worked for me  while including that compatibility warning.
However, I propose a split of extension versions as well, as there is possibly no change for the old visual studio versions to come.

MFG
HeikoStudt

**Comment by pasztorpisti** [comment]
@mdvtj @HeikoStudt

Hi guys!

Thank you for doing the research!

In my opinion dropping support for older IDEs (especially the 7 years VS2010) isn't a huge guilt. Those who want this (old and buggy :-D hehe) plugin for an older IDE can download and dblclick+install it "manually" from here: https://github.com/pasztorpisti/vs-window-title-changer/blob/master/Builds/r112/VSWindowTitleChanger.vsix (These plugins should really be in the release download section of this github repo but I've imported this codebase from an ancient svn repo...)

If someone updates this codebase for VS2017 then credits can be put into the help file somewhere near to the copyright text: https://github.com/pasztorpisti/vs-window-title-changer/blob/master/Forms/TitleSetupEditorHelp.html

You can add multiple people to the credits, especially because you can do a better job at reviewing the new changes - I don't have a windows devenv and my C# dev skills are probably rusty.

The README.md of the repo could also be updated:
- Pointing people with older IDEs to the old plugin binary
- Adding the credits

**Comment by mdvtj** [comment]
I have attached a build of a migrated version ([VSWindowTitleChanger.zip](https://github.com/pasztorpisti/vs-window-title-changer/files/985522/VSWindowTitleChanger.zip)). See this [commit](https://github.com/mdvtj/vs-window-title-changer/commit/f3bf0edb5d26ea745c180de323c95ecdb5fa6df0) for more details.

I have tested this with VS2017 and VS2015. In theory, older versions including VS2012 should be still supported. Feedback if this works for others is welcome. I would be happy to prepare a pull request after some more days of testing in production.

**Comment by pasztorpisti** [comment]
@mdvtj Thank you for your help.

I've tried it on my friend's windows machine with VS2017 Community Edition and it crashed immediately when I started editing the title expression. This isn't your fault, it's my messy and buggy multithreaded code.

I've fixed the crash bug and compiled a new vsix binary - it's available as v2.1.8 in the VS Gallery. I've reused your solution & manifest files in the process.
